src/kv-queue/module.ts

```diff
@@ -0,0 +1,55 @@
+import { OnAppBootstrap, OnAppClose } from '../hook/interfaces.ts';
+import { MetadataHelper } from '../metadata/helper.ts';
+import { injector, Logger, Module } from '../mod.ts';
+import { queueListenerMetadataKey } from './constants.ts';
+import { KvQueue } from './kv.ts';
+
+@Module({})
+export class KvQueueModule implements OnAppBootstrap, OnAppClose {
+	private logger: Logger = new Logger('QueueModule');
+
+	public injectables = [KvQueue];
+
+	constructor(private kvName?: string) {}
+
+	public static forRoot(kvName?: string) {
+		return new KvQueueModule(kvName);
+	}
+
+	async onAppBootstrap(): Promise<void> {
+		const queue = injector.get<KvQueue>(KvQueue);
+		await queue.start(this.kvName);
+		for (const instance of injector.injectables) {
+			this.registerAvailableEventListeners(instance);
+		}
+	}
+
+	onAppClose() {
+	}
+
+	// deno-lint-ignore no-explicit-any
+	private registerAvailableEventListeners(injectableInstance: any) {
```

Comment:
(no change required) FYI, I'm considering a future ref for sharing this logic between our modules. LMK your thoughts about the best way of doing this 🙏🏼